mail/parser.go

```diff
@@ -174,14 +174,15 @@ func (parser *Parser) parseMultipartSigned(contentType *MimeMediaType, header te
 	if len(result.Parts) != 2 {
 		return nil, errors.New("multipart/signed mail must contain exactly two parts")
 	}
-	signatureHeader := result.Parts[1].getHeader("Content-Type", "")
-	if signatureHeader != "application/pgp-signature" {
+	// Because we already successfully parsed the multipart content, no error can be returned here.
+	signatureHeader, _ := getMimeMediaTypeFromHeader(result.Parts[1].Header, "Content-Type", "")
```

Comment:
"no error can be returned here" is a bold statement, I would prefer to still check the returned error value and call `panic("should never happen")` if not `nil`.